timemachine/cpp/src/kernels/k_periodic_torsion.cuh

```diff
@@ -2,16 +2,16 @@
 #include "k_fixed_point.cuh"
 
 template <typename RealType> inline __device__ RealType dot_product(const RealType a[3], const RealType b[3]) {
-
     return a[0] * b[0] + a[1] * b[1] + a[2] * b[2];
 }
 
 template <typename RealType>
 inline __device__ void cross_product(const RealType a[3], const RealType b[3], RealType c[3]) {
-    // fix this one indexed garbage later
-    c[1 - 1] = a[2 - 1] * b[3 - 1] - a[3 - 1] * b[2 - 1];
-    c[2 - 1] = a[3 - 1] * b[1 - 1] - a[1 - 1] * b[3 - 1];
-    c[3 - 1] = a[1 - 1] * b[2 - 1] - a[2 - 1] * b[1 - 1];
+    // these extra __dmul_rn calls are needed to preserve bitwise anticommutativity
+    // i.e. cross(a,b) is bitwise identical to -cross(b,a) except in the sign-bit
+    c[0] = __dmul_rn(a[1], b[2]) - __dmul_rn(a[2], b[1]);
```

Comment:
Should this always be the double-precision intrinsic `__dmul_rn` regardless of `RealType`?

Reply:
great catch! d'oh.